### ISSUE TITLE
Official VSCode/ium extension

### ISSUE THREAD

**Issue by cab404** [direnv/direnv#844]
**Is your feature request related to a problem? Please describe.**
I am kind of frustruated, that there is no "direnv" extension for vscode.
Recently I found out that apart from rubymaniac.direnv and cab404.direnv (fork of latter), there is mkhl.direnv.
It gets out of control :D

**Describe the solution you'd like**
Make an official one. Either select whichever you want and ask to move it into direnv (like nix-community does), or just write one from scratch, since that's a new trend :D

At least reserve "direnv" namespaces on both openvsx and vscode store.

**Comment by zimbatm** [comment]
I'm not super aware of the space. Did rubymaniac's extension stop being supported, and how does the mkhl.direnv version relate to your fork?

I'm happy to host an "official" version, if it gets supported by the community.

**Comment by cab404** [comment]
@rubymaniac disappeared, and extension broke. So I've decided to maintain it, and since there were no way to push updates to original extension, I've created a new one.
After a while, @mkhl suddenly appeared with a [nice suggestion in an issue in my repo](https://github.com/cab404/vscode-direnv/issues/6#issuecomment-976927640), and it turned out he too has a direnv extension, created on 24.11.2021 :D
So, yeah, that's a small recap of what's happening in this field.

**Comment by cab404** [comment]
Since both of those extensions are really tiny, it wouldn't be hard to squash source of both. 
By the looks of it, @mkhl has a more modern and nice approach to modifying environment, however I didn't yet test it with extensions relying on executables in PATH. If that works, his source should be better.  
@mkhl uses some super-short version of MIT I don't recognize, and I use MIT from rubymaniacs repo.

**Comment by cab404** [comment]
Of course, it would be nice to also hear @mkhl's opinion on the matter

**Comment by zimbatm** [comment]
Agreed. It's really cool to see you both invested in this by the way. Let's wait for what mkhl has to say but I like the sound of what you're proposing.

**Comment by mkhl** [comment]
yes hi sorry

i agree that both extensions shouldn't be hard to merge, and that my tree would probably make that easier.
and i'd be happy to work on making an "official" version.

fwiw from my side the story is that i tried using the old direnv extensions but they didn't support my use case (cab404/vscode-direnv#6),
and when i stumbled on a solution it seemed easier to create a new extension for it.

i feel i should have talked to you about all that beforehand, sorry that i didn't.

> however I didn't yet test it with extensions relying on executables in PATH. If that works, his source should be better

in my brief test it seemed to work, did you get a chance to test this yet?

> mkhl uses some super-short version of MIT I don't recognize

that's the ["Zero Clause BSD License"](https://spdx.org/licenses/0BSD.html).
i like it for its intention (being a public domain equivalent license) and it's brevity.

**Comment by cab404** [comment]
I've just took some time to check it — by the looks of it it does not work with rust-analyzer extension
I would guess adding blocking behavior should fix that, but I want to see, whether I can fix that in rust-analyzer itself (dunno, maybe there is some way to await environmentvariablecollection?)

**Comment by mkhl** [comment]
> edit: rust-analyzer doesn't even check EVC. So I guess we'll need editing `env`

i just released v0.1.0 which does this, could you check if it works for you?

**Comment by cab404** [comment]
> > edit: rust-analyzer doesn't even check EVC. So I guess we'll need editing `env`
> 
> i just released v0.1.0 which does this, could you check if it works for you?

Not really. It looks like it RAL doesn't wait for anything to load, and just goes ahead and loads PATH in front of direnv.
Are you blocking the main thread before getting variables into `process.env`? Cause that's the only way I found to actually get variables in before anything activates.

**Comment by cab404** [comment]
I can probably PR, if you don't have time for that.
@zimbatm can you please clone https://codeberg.org/mkhl/vscode-direnv/ in direnv/vscode?

**Comment by mkhl** [comment]
> Are you blocking the main (extension) thread before getting variables into process.env? Cause that's the only way I found to actually get variables in before anything activates.

oh darn, no i'm not. that's activating on `*` and awaiting the first reload promise on the main thread right?

**Comment by cab404** [comment]
[Yup](https://github.com/cab404/vscode-direnv/blob/master/src/main.ts#L154). Hacky as hell :(

**Comment by cab404** [comment]
We can probably do direnv caching in somewhere in VSCode and just apply the latest cache on reload, and just prompt to reload if cached version does not match loaded one.

**Comment by mkhl** [comment]
> Not really. It looks like it RAL doesn't wait for anything to load, and just goes ahead and loads PATH in front of direnv.

i might be missing something but the rust analyzer extension doesn't seem to check PATH at all? (see https://github.com/rust-analyzer/rust-analyzer/blob/d9b2291f546abc77d24499339a72a89127464b95/editors/code/src/main.ts#L311)

**Comment by cab404** [comment]
> > Not really. It looks like it RAL doesn't wait for anything to load, and just goes ahead and loads PATH in front of direnv.
> 
> i might be missing something but the rust analyzer extension doesn't seem to check PATH at all? (see https://github.com/rust-analyzer/rust-analyzer/blob/d9b2291f546abc77d24499339a72a89127464b95/editors/code/src/main.ts#L311)

It uses it with cargo.

**Comment by zimbatm** [comment]
I created an empty repo and a team, to which you are both invited. There is a Discussions page you can use to coordinate your work here: https://github.com/orgs/direnv/teams/vscode/discussions
